File: python/max_consecutive_ones.py
# https://leetcode.com/explore/learn/card/fun-with-arrays/521/introduction/3238/

class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        max_seq = 0
        cur_seq = 0
        for num in nums:
            if num == 1:
                cur_seq += 1
            else:
                if cur_seq > max_seq:
                    max_seq = cur_seq
                cur_seq = 0
        
        return max(cur_seq, max_seq)

File: leetcode/java/MaxConsecutiveOnes.java
// https://leetcode.com/explore/learn/card/array-and-string/205/array-two-pointer-technique/1301/

// O(n)
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        boolean cons = false;
        int n = 0, N = 0;
        
        for(int i : nums) {
            if(i == 1) {
                n++;
                cons = true;
            } else {
                if(cons) {
                    if(n > N) {
                        N = n;
                    }
                }
                cons = false;
                n = 0;
            }
        }
        
        return N > n ? N : n;
    }
}
